### ISSUE TITLE
Raw batch reading for ring buffers

### ISSUE THREAD

**Issue by Scooletz** [AdaptiveConsulting/Aeron.NET#62]
Would you consider a PR introducing a batched read for ring buffers, accessing the buffer directly and passing it to a handler? Something similar to https://github.com/Scooletz/RampUp/blob/feature/read-raw-chunk/src/RampUp/Ring/ManyToOneRingBuffer.cs#L129-L179

This would allow to use ring buffer as a "concurrent queue that can be read in batches" and clear the memory afterwards.

**Comment by JPWatson** [comment]
something like this?https://github.com/AdaptiveConsulting/Aeron.NET/blob/master/src/Adaptive.Agrona/Concurrent/RingBuffer/ManyToOneRingBuffer.cs#L114

**Comment by Scooletz** [comment]
This is not what I meant. Let me rephrase my initial description.
The `Read` method provides a capability of reading multiple messages one after another and clears the memory afterwards. It does not provide an access to the whole chunk of memory though. I could imagine scenarios, where one wants to have multiple writers which write messages to a buffer. Then a reader would access the whole chunk of buffer and use it somehow (send to a file/socket/something else). In this case existing `Read` wouldn't be suitable - it reads messages one after another.

**Comment by Scooletz** [comment]
That's correct :+1: I'm thinking now, is it the right lvl for introducing it?

**Comment by JPWatson** [comment]
If you Read using the MessageHandler, you could capture buffer & the first index then sum up the lengths of all the individual messages. Then you have access to the whole block.

**Comment by Scooletz** [comment]
You can't do it because in any point of time (at the end) cleaning memory and moving head can be called. That's why I asked about making this explicit and calling the "raw handler" only one time.

**Comment by JPWatson** [comment]
the position is only advanced after the whole batch has been processed 
https://github.com/AdaptiveConsulting/Aeron.NET/blob/master/src/Adaptive.Agrona/Concurrent/RingBuffer/ManyToOneRingBuffer.cs#L152

**Comment by JPWatson** [comment]
hmm I take that back, you can't do that unless you know the last message in the batch

**Comment by JPWatson** [comment]
While I see the benefit of the request, I'm hesitant to introduce changes to this code base that aren't used directly by the Aeron client. It seems like you've already got what you need in RampUp. 
If you haven't already, it might be worth having a look at the [Disruptor.NET](https://github.com/disruptor-net/Disruptor-net).
Looking at the EventHandler in [Getting started (java)](https://github.com/LMAX-Exchange/disruptor/wiki/Getting-Started) it seems like it might fit with your use case.

**Comment by Scooletz** [comment]
Thx for clarification. Closing this :)
